README.md

```diff
@@ -210,51 +210,74 @@ Used for implementing pagination or infinite scroll (load more items when the us
 
 In order to intercept clicks, you should implement the interface `com.taboola.android.api.TaboolaOnClickListener` and set it in the sdk.
 
-```java
-   TaboolaApi.getInstance().setOnClickListener(new TaboolaOnClickListener() {
-       @Override
-       public boolean onItemClick(String placementName, String itemId, String clickUrl, boolean isOrganic) {
-           return false;
-       }
-   });
-
-```
-
-This method will be called every time a user clicks a recommendation, right before triggering the default behavior. You can block default click handling for organic items by returning `false` in `onItemClick()` method.
-
-* Return **`false`** - abort the default behavior, the app should display the recommendation content on its own (for example, using an in-app browser). (Aborts only for organic items!)
-* Return **`true`** - this will allow the app to implement a click-through and continue to the default behaviour.
-
-`isOrganic` indicates whether the item clicked was an organic content recommendation or not.
-**Best practice would be to suppress the default behavior for organic items, and instead open the relevant screen in your app which shows that piece of content.**
-
+1. Implement the interface `com.taboola.android.api.TaboolaOnClickListener` 
+    1.1 `TaboolaOnClickListener` include the methods:
+     ```java
+    public boolean onItemClick(String placementName, String itemId, String clickUrl, boolean isOrganic);
+     ```
+    1.2 Example implementation:
+    In the same Activity/Fragment as `TaboolaWidget` instance:
+     ```java
+    TaboolaOnClickListener taboolaOnClickListener = new TaboolaOnClickListener() {
+      @Override
+      public boolean onItemClick(String placementName, String itemId, String clickUrl, boolean isOrganic) {          
+          //Code...
+          return false;
+      }};
+     ```    
+2. Connect the event listener to your `TaboolaWidget` instance. 
+    ```java
+    TaboolaApi.getInstance().setOnClickListener(taboolaOnClickListener);
+    ```    
+    
+##### 1.9.3. Event: taboolaViewItemClickHandler
+`boolean onItemClick(String placementName, String itemId, String clickUrl, boolean isOrganic)`
+This method will be called every time a user clicks on a Taboola Recommendation, right before it is sent to Android OS for relevant action resolve. The return value of this method allows you to control further system behaviour (after your own code executes).
+
+###### 1.9.3.1 `placementName:`
+The name of the placement, in which an Item was clicked.
+
+###### 1.9.3.2 `itemtId:`
+The id of the Item clicked.
+
+###### 1.9.3.3 `clickUrl:`
+Original click url.
+
+###### 1.9.3.4 `isOrganic:` 
+Indicates whether the item clicked was an organic content Taboola Recommendation or not.
+(The **best practice** would be to suppress the default behavior for organic items, and instead open the relevant screen in your app which will show that piece of content).
```

Comment:
It's best practice, it depends on the way the publisher wants to handle organic click